### PR TITLE
Sort by gross margin per hour

### PR DESCRIPTION
**Pull request by bryceholcomb**

### CODE REVIEW

File: app/models/appointment.rb [hunk 1]
```diff
@@ -48,9 +48,9 @@ def gross_margin
     appointment_value - estimated_cogs if appointment_value
   end
 
-  def gross_margin_per_hour
-    gross_margin / length_in_hours if gross_margin
-  end
+  #def gross_margin_per_hour
```
Comment (bryceholcomb) [inline app/models/appointment.rb]: Will remove this after we decide everything is working as it should

File: lib/tasks/data.rake [hunk 2]
```diff
@@ -26,6 +26,12 @@ namespace :data do
 
       total_hours = appointments.map(&:length_in_hours).inject(0, &:+)
       job.update(total_hours: total_hours)
+      job.appointments.each do |appt|
```
Comment (bryceholcomb) [inline lib/tasks/data.rake]: @mrhamric looks like I can scope this down to just active appointments as those are the only ones being displayed. Does that sound correct to you?

Comment (mrhamric) [reply lib/tasks/data.rake]: Yep, that's correct. IIRC there's a scope the model you can plug in here

File: db/schema.rb [hunk 3]
```diff
@@ -32,10 +32,7 @@
     t.integer "job_id"
     t.integer "payroll_period_id"
     t.integer "technician_id"
-    t.index ["appointment_start"], name: "index_appointments_on_appointment_start"
-    t.index ["payroll_period_id"], name: "index_appointments_on_payroll_period_id"
-    t.index ["status"], name: "index_appointments_on_status"
-    t.index ["vendor"], name: "index_appointments_on_vendor"
```
Comment (mrhamric) [inline db/schema.rb]: weird, ill fix this on merge

File: lib/tasks/data.rake [hunk 4]
```diff
@@ -26,6 +26,12 @@ namespace :data do
...
+        if appt.gross_margin
+          gross_margin_per_hour = appt.gross_margin / appt.length_in_hours
+          appt.update(gross_margin_per_hour: gross_margin_per_hour)
```
Comment (mrhamric) [inline lib/tasks/data.rake]: im kinda thinking I'd like to have this logic somewhere else in the app, as we use it all over the place, but I'm not sure it belongs on the appointment anymore. definitely not a dealbreaker either way, what do you think?

**Pull request merged**